homeassistant/components/fivem/__init__.py

```diff
@@ -0,0 +1,194 @@
+        was_online = self.online
+
+        try:
+            server = await self._fivem.get_server()
+            self.online = True
+        except FiveMServerOfflineError:
+            self.online = False
+
+        if was_online and not self.online:
+            _LOGGER.warning("Connection to '%s:%s' lost", self.host, self.port)
+        elif not was_online and self.online:
+            _LOGGER.info("Connection to '%s:%s' (re-)established", self.host, self.port)
+
+        if self.online:
+            players_list: list[str] = []
+            for player in server.players:
+                players_list.append(player.name)
+            players_list.sort()
+
+            resources_list = server.resources
+            resources_list.sort()
+
+            return {
+                NAME_PLAYERS_ONLINE: len(players_list),
+                NAME_PLAYERS_MAX: server.max_players,
+                NAME_RESOURCES: len(resources_list),
+                NAME_STATUS: self.online,
+                ATTR_PLAYERS_LIST: players_list,
+                ATTR_RESOURCES_LIST: resources_list,
+            }
+
+        raise UpdateFailed
+
+
+class FiveMEntity(CoordinatorEntity):
+    """Representation of a FiveM base entity."""
+
+    coordinator: FiveMDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: FiveMDataUpdateCoordinator,
+        type_name: str,
+        icon: str,
+        device_class: str = None,
+        extra_attrs: list[str] = None,
+    ) -> None:
+        """Initialize base entity."""
+        super().__init__(coordinator)
+        self.type_name = type_name
+        self.extra_attrs = extra_attrs
+
+        self._attr_name = f"{self.coordinator.server_name} {type_name}"
+        self._attr_unique_id = f"{self.coordinator.unique_id}-{type_name}".lower()
+        self._attr_icon = icon
+        self._attr_device_class = device_class
+        self._attr_should_poll = False
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, self.coordinator.unique_id)},
+            "manufacturer": MANUFACTURER,
+            "model": self.coordinator.server,
+            "name": self.coordinator.server_name,
+            "sw_version": self.coordinator.version,
+        }
+
+        self._update_entity()
+
+    def _update_entity(self):
+        """Update the entity."""
+        self._update_value()
+
+        if self.extra_attrs is not None:
+            extras: dict[str, Any] = {}
+            for attr in self.extra_attrs:
+                extras[attr] = self.coordinator.data[attr]
+            self._attr_extra_state_attributes = extras
+
+    @abstractmethod
+    def _update_value(self):
+        """Update the value of the entity."""
+        return NotImplemented
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Triggers update of properties after receiving update from coordinator."""
+        self._update_entity()
+        self.async_write_ha_state()
```

Comment:
Why do you need this? Is the default `_handle_coordinator_update` from `CoordinatorEntity` not enough? See my other related comment as well :).

Reply:
I removed this part of the code and used a property just like the `native_value` and `is_on`